### DISCUSSION TITLE
OGCAPI Records as alternative discovery service protocol

### DISCUSSION THREAD

**Discussion by pvgenuchten** [INSPIRE-MIF/helpdesk#123] [Ideas]
Are there initiatives to craft a good practice document for using OGCAPI Records as alternative discovery service protocol?

A good practice document for download service via OGCAPI Features already exists, endorsing adoption of OGCAPI Records would also make sense in my opinion.

Activities envisioned within this good practice
- Add OGCAPI Records as a discovery protocol in the [INSPIRE registry](https://inspire.ec.europa.eu/metadata-codelist/ProtocolValue)
- Either require the iso19139 xml format as encoding of the getrecord-api-call, or extend the [record format](https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/recordGeoJSON.yaml) to require dedicated INSPIRE properties
- Add capability to INSPIRE GeoPortal Dashboard to configure an OGCAPI Records endpoint to be harvested by the portal
- Add test to ETF validator for OGCAPI Records endpoint
- Verify/Describe service linkage between OGCAPI Record and known download and view services and alternate metadata linkage

**Comment by reinij** [comment]
Hi,

Not yet. At least at this stage, the adoption of the standard is awaited. Some tests are likely to be done in the GeoE3 project.

If you have the enthusiasm to start the process of creating OGC API Records as an alternative way of implementing discovery service, I am interested in participating in this.

**Comment by pvgenuchten** [comment]
Thanx for the comment jari, yes I’d be interested to contribute to this activity, can I make a pull request with a draft document to this repository?

**Comment by MarcoMinghini** [comment]
Thanks @pvgenuchten and @reinij for the interest in strating a new Good Practice document!

We (JRC) are also very interested to bring the new standard onboard - at the moment we do not have the bandwidth to lead the activity, but for sure we can keep an eye open and help when needed.

Each Good Practice should be placed in a separate repository, see e.g. [the one for OGC API - Features](https://github.com/INSPIRE-MIF/gp-ogc-api-features/). If this is ok, I can create a new repository and add you both as collaborators.

**Comment by pvgenuchten** [comment]
Thank you Marco, that would be helpful! Also curious to hear from you if you'd like us to include #65 or keep it seperately, the topic may grow too big, and also csw could benefit from a dcat schema.

I made a start for such a repo at https://github.com/pvgenuchten/gp-ogc-api-records, as you can see i'm heavily inspired by the great gp ogcapi features work...

**Comment by MarcoMinghini** [comment]
Dear @pvgenuchten and @reinij, I created [this new repository](https://github.com/INSPIRE-MIF/gp-ogc-api-records) and gave you write access.

The topic of #65 is of high interest, however I am not sure if it should be part of a GP on OGC API - Records or if it should be an additional one (although they are of course related).

Please use the issue tracker of the new repository for all discussions (I can move issues from this repo to that repo, if useful) and more in general feel free to use that space for content, ideas, etc. 

Only some warnings:
* the standard should be adopted before it can become an INSPIRE GP (now it is still a draft)
* the idea of [Good Practices](https://inspire.ec.europa.eu/portfolio/good-practice-library) is that proposals are supported by a wide community, with implementation evidence already existing. So this is probably the first step to do - involve other implementers and early adopters and start discussing the scope and the details together with them. 

Thanks for this initiative, we will be happy to stay involved.
